connectors/es/index.py

```diff
@@ -50,10 +62,10 @@ async def get_all_docs(self, query=None, page_size=DEFAULT_PAGE_SIZE):
         Returns:
             Iterator
         """
-        await self.client.indices.refresh(index=self.index_name)
-
         if query is None:
-            query = {"match_all": {}}
+            return
```

Comment:
if query has to be provided, let's make it a mandatory argument

Reply:
Even it's mandatory, it's still possible to be `None`.

Reply:
Yes, or -1 or any bad value because we don’t assert we get a query. But in practice if the argument is mandatory the expectation is to get a query. Which is what we do here because if we don’t get one we return and fail in practice. Query validation is another topic.